app/models/sample_manifest/core_behaviour.rb

```diff
@@ -37,5 +37,5 @@ def core_behaviour
     core = rapid_generation? ? 'RapidCore' : 'Core'
     @core_behaviour = "::SampleManifest::#{behaviour}::#{core}".constantize.new(self)
   end
```

Comment:
Be a bit wary of making public methods private. Instead think about what your actually wanting to expose and make that public. From looking through, it appears your mostly accessing the tubes/plates/samples. We'll step through how you might approach this one together, as I think it'll help clean up some of the code elsewhere as well.